examples/nix_cross_compiling/BUILD.bazel

```diff
@@ -0,0 +1,167 @@
+load("@aspect_bazel_lib//lib:transitions.bzl", "platform_transition_binary")
```

Comment:
Can this instead be implemented in this example? I'd love to reduce dependencies where possible and this transition should be pretty simple.

Reply:
I'm hoping to make (very) soon is rust_binary() (and whatever Rust dylib is, etc.) take an optional platform transition.  Given Bazel's lack of "rule wrappers", platform_transition_binary() causes a whole bunch of headaches with things like aspects, bazel test, etc.  If it's okay, I think it would be easier to ship this as is, and then remove it in a follow up PR with that change (if you're not opposed to having an optional transition on those rules).

Reply:
I think it'd still be better to add a `.bzl` file here
```starlark
"""Rules for working with transitions."""

load("@bazel_skylib//lib:paths.bzl", "paths")

def _transition_platform_impl(_, attr):
    return {"//command_line_option:platforms": str(attr.target_platform)}

# Transition from any input configuration to one that includes the
# --platforms command-line flag.
_transition_platform = transition(
    implementation = _transition_platform_impl,
    inputs = [],
    outputs = ["//command_line_option:platforms"],
)

def _platform_transition_binary_impl(ctx):
    # We need to forward the DefaultInfo provider from the underlying rule.
    # Unfortunately, we can't do this directly, because Bazel requires that the executable to run
    # is actually generated by this rule, so we need to symlink to it, and generate a synthetic
    # forwarding DefaultInfo.

    result = []
    binary = ctx.attr.binary[0]

    default_info = binary[DefaultInfo]
    files = default_info.files
    new_executable = None
    original_executable = default_info.files_to_run.executable
    runfiles = default_info.default_runfiles

    if not original_executable:
        fail("Cannot transition a 'binary' that is not executable")

    new_executable_name = original_executable.basename

    # In order for the symlink to have the same basename as the original
    # executable (important in the case of proto plugins), put it in a
    # subdirectory named after the label to prevent collisions.
    new_executable = ctx.actions.declare_file(paths.join(ctx.label.name, new_executable_name))
    ctx.actions.symlink(
        output = new_executable,
        target_file = original_executable,
        is_executable = True,
    )
    files = depset(direct = [new_executable], transitive = [files])
    runfiles = runfiles.merge(ctx.runfiles([new_executable]))

    result.append(
        DefaultInfo(
            files = files,
            runfiles = runfiles,
            executable = new_executable,
        ),
    )

    return result

platform_transition_binary = rule(
    doc = "Transitions the binary to use the provided platform.",
    implementation = _platform_transition_binary_impl,
    attrs = {
        "binary": attr.label(
            doc = "The target to transition.",
            allow_files = True,
            cfg = _transition_platform,
            mandatory = True,
        ),
        "target_platform": attr.label(
            doc = "The target platform to transition the binary.",
            mandatory = True,
        ),
        "_allowlist_function_transition": attr.label(
            default = "@bazel_tools//tools/allowlists/function_transition_allowlist",
        ),
    },
    executable = True,
)
```
I would add this to `examples/nix_cross_compiling/transition.bzl` and load that here instead to eliminate a dependency and keep folks from needing to jump around to other github repos when looking at this example